src/Temporalio/Client/Schedules/ScheduleUpdate.cs

```diff
@@ -1,8 +1,25 @@
+using Temporalio.Common;
+
 namespace Temporalio.Client.Schedules
 {
     /// <summary>
     /// An update returned from an updater.
     /// </summary>
     /// <param name="Schedule">Schedule to update.</param>
-    public record ScheduleUpdate(Schedule Schedule);
+    /// <param name="SearchAttributes">Optional indexed attributes that can be used while querying
+    /// schedules via the list schedules APIs. The key and value type must be registered with
+    /// Temporal server.
+    ///
+    /// <ul>
+    /// <li>If null, the search attributes will not be updated.</li>
+    /// <li>If present but empty, the search attributes will be cleared.</li>
+    /// <li>If present and non-empty, all search attributes will be updated to the provided
+    ///     collection. This means existing attributes which do not exist in the provided collection
+    ///     will be wiped out. You can copy attributes from the existing <see
+    ///     cref="ScheduleDescription"/> to avoid this.</li>
+    /// </ul>
+    /// </param>
+    public record ScheduleUpdate(
+        Schedule Schedule,
+        SearchAttributeCollection? SearchAttributes = null);
```

Comment:
```suggestion
        SearchAttributeCollection? TypedSearchAttributes = null);
```

This matches the name on `ScheduleOptions` when creating. I considered accepting `ScheduleOptions` when the inevitable memo update comes about, but it seemed more bad than good.